code/modules/power/cable.dm

```diff
@@ -472,12 +473,28 @@ var/global/list/datum/stack_recipe/cable_coil_recipes = list ( \
 	item_color = cable_color
 	update_icon()
 
-/obj/item/stack/cable_coil/suicide_act(mob/user)
-	if(locate(/obj/structure/chair/stool) in get_turf(user))
-		user.visible_message("<span class='suicide'>[user] is making a noose with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+/obj/item/stack/cable_coil/suicide_act(mob/living/user)
+	if((locate(/obj/structure/bed) in usr.loc) || (locate(/obj/structure/table) in user.loc) || (locate(/obj/structure/toilet) in user.loc))

```

Comment:
what the fuck is this, use get_turf(user) once and locate on the resulting turf object, also make sure that the value from get_turf is not null